server/lessor.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import (
+	"github.com/coreos/etcd/clientv3"
+	concurv3 "github.com/coreos/etcd/clientv3/concurrency"
+	"github.com/juju/errors"
+	"github.com/pingcap/kvproto/pkg/pdpb"
+	"golang.org/x/net/context"
+)
+
+var (
+	errNoLeader = errors.New("no leader elected")
+)
+
+// Lessor is a wrapper for leader election.
+type Lessor struct {
+	session  *concurv3.Session
```

Comment:
concurv3 seems not readable (concur is another word). Why not using concurrency directly?